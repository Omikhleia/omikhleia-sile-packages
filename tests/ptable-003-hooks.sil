\begin{document}
\script[src=packages/ptable]

% Simple table (same as test 002) with additional cell hooks

\script{
  -- redefine the ptable:cell:hook no-op command to do something
  SILE.registerCommand("ptable:cell:hook", function(options, content)
    if options.style == "center" then
      SILE.call("center", {}, content)
    else
      SILE.process(content)
    end
  end)
}

\begin[cols=20%fw 15%fw 15%fw 50%fw]{ptable}
  \begin[background=#ddd]{row}
    \cell{Day}
    \begin[span=3]{celltable}
      \begin{row}
        \cell[span=3, style=center]{Seminar}
      \end{row}
      \begin{row}
        \begin[span=2]{celltable}
          \begin{row}
            \cell[span=2, style=center]{Schedule}
          \end{row}
          \begin{row}
            \cell[style=center]{Begin}
            \cell[style=center]{End}
          \end{row}
        \end{celltable}
        \cell{Topic}
      \end{row}
    \end{celltable}
  \end{row}
  \begin{row}
    \cell[background=#eee]{Monday}
    \cell[style=center]{9:30 AM}
    \cell[style=center]{5:30 PM}
    \celltable{
        \row{\cell{Introduction to SILE}}
        \row{\cell{Step by step: classes & packages}}
    }
  \end{row}
    \begin{row}
    \cell[background=#eee]{Tuesday}
    \celltable[span=2]{
        \row{\cell[style=center]{9:30 AM}\cell[style=center]{11:00 AM}}
        \row{\cell[style=center]{11:00 AM}\cell[style=center]{12:00 PM}}
        \row{\cell[style=center]{12:00 PM}\cell[style=center]{5:00 PM}}
    }
    \celltable{
        \row{\cell{Math & advanced typesetting}}
        \row{\cell{The internals of SILE}}
    }
  \end{row}
  \begin{row}
    \cell[background=#eee]{Wednesday}
    \cell[style=center]{8:00 AM}
    \cell[style=center]{12:00 PM}
    \cell{SILE as an XML processor}
  \end{row}
\end{ptable}

\end{document}