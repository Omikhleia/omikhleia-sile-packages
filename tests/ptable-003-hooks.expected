Set paper size 	297.6377985	419.5275636
Begin page
Push color	0.8667	0.8667	0.8667
Draw line	14.8819	20.9764	267.8740	52.6084
Pop color
Mx 	14.8819
My 	47.2806
Draw line	14.8819	20.9764	53.5748	0.4000
Draw line	14.8819	73.5848	53.5748	0.4000
Draw line	14.8819	20.9764	0.4000	53.0084
Draw line	68.4567	20.9764	0.4000	53.0084
Mx 	18.8819
My 	49.8514
Set font 	Gentium Plus;9;400;;normal;;LTR
T	39 68 92 w=13.9043	(Day)
Mx 	68.4567
My 	29.7444
Draw line	68.4567	20.9764	214.2992	0.4000
Draw line	68.4567	38.5125	214.2992	0.4000
Draw line	68.4567	20.9764	0.4000	17.9361
Draw line	282.7559	20.9764	0.4000	17.9361
Mx 	160.2210
My 	32.3152
T	54 72 80 76 81 68 85 w=30.7705	(Seminar)
Mx 	68.4567
My 	56.0486
My 	47.2806
Draw line	68.4567	38.5125	85.7197	0.4000
Draw line	68.4567	56.0486	85.7197	0.4000
Draw line	68.4567	38.5125	0.4000	17.9361
Draw line	154.1764	38.5125	0.4000	17.9361
Mx 	94.6239
My 	49.8514
T	54 70 75 72 71 88 79 72 w=33.3853	(Schedule)
Mx 	68.4567
My 	64.8167
Draw line	68.4567	56.0486	42.8598	0.4000
Draw line	68.4567	73.5848	42.8598	0.4000
Draw line	68.4567	56.0486	0.4000	17.9361
Draw line	111.3165	56.0486	0.4000	17.9361
Mx 	79.4364
My 	67.3875
T	37 72 74 76 81 w=20.9004	(Begin)
Draw line	111.3165	56.0486	42.8598	0.4000
Draw line	111.3165	73.5848	42.8598	0.4000
Draw line	111.3165	56.0486	0.4000	17.9361
Draw line	154.1764	56.0486	0.4000	17.9361
Mx 	125.6910
T	40 81 71 w=14.1108	(End)
Draw line	154.1764	38.5125	128.5795	0.4000
Draw line	154.1764	73.5848	128.5795	0.4000
Draw line	154.1764	38.5125	0.4000	35.4723
Draw line	282.7559	38.5125	0.4000	35.4723
Mx 	158.1764
My 	58.6194
T	55 82 83 76 70 w=20.6235	(Topic)
Mx 	14.8819
My 	91.1209
Push color	0.9333	0.9333	0.9333
Draw line	14.8819	73.5848	53.5748	35.0723
Pop color
Draw line	14.8819	73.5848	53.5748	0.4000
Draw line	14.8819	108.6570	53.5748	0.4000
Draw line	14.8819	73.5848	0.4000	35.4723
Draw line	68.4567	73.5848	0.4000	35.4723
Mx 	18.8819
My 	93.6917
T	48 82 81 71 68 92 w=29.9971	(Monday)
Draw line	68.4567	73.5848	42.8598	0.4000
Draw line	68.4567	108.6570	42.8598	0.4000
Draw line	68.4567	73.5848	0.4000	35.4723
Draw line	111.3165	73.5848	0.4000	35.4723
Mx 	75.0261
T	28 w=4.2231	(9)
Mx 	79.2492
T	29 w=2.0610	(:)
Mx 	81.3102
T	22 19 w=8.4463	(30)
Mx 	92.1349
T	36 48 w=12.6123	(AM)
Draw line	111.3165	73.5848	42.8598	0.4000
Draw line	111.3165	108.6570	42.8598	0.4000
Draw line	111.3165	73.5848	0.4000	35.4723
Draw line	154.1764	73.5848	0.4000	35.4723
Mx 	118.2111
T	24 w=4.2231	(5)
Mx 	122.4343
T	29 w=2.0610	(:)
Mx 	124.4953
T	22 19 w=8.4463	(30)
Mx 	135.3199
T	51 48 w=11.9619	(PM)
Mx 	154.1764
My 	82.3528
Draw line	154.1764	73.5848	128.5795	0.4000
Draw line	154.1764	91.1209	128.5795	0.4000
Draw line	154.1764	73.5848	0.4000	17.9361
Draw line	282.7559	73.5848	0.4000	17.9361
Mx 	158.1764
My 	84.9236
T	44 81 87 85 82 71 88 70 87 76 82 81 w=47.2456	(Introduction)
Mx 	207.8048
T	87 82 w=7.6245	(to)
Mx 	217.8121
T	54 44 47 40 w=15.7148	(SILE)
Mx 	154.1764
My 	99.8890
Draw line	154.1764	91.1209	128.5795	0.4000
Draw line	154.1764	108.6570	128.5795	0.4000
Draw line	154.1764	91.1209	0.4000	17.9361
Draw line	282.7559	91.1209	0.4000	17.9361
Mx 	158.1764
My 	102.4598
T	54 87 72 83 w=16.3081	(Step)
Mx 	176.8629
T	69 92 w=9.0088	(by)
Mx 	188.2502
T	86 87 72 83 w=15.4995	(step)
Mx 	203.7497
T	29 w=2.0610	(:)
Mx 	208.1892
T	70 79 68 86 86 72 86 w=25.0884	(classes)
Mx 	235.6560
T	9 w=6.0029	(&)
Mx 	244.0374
T	83 68 70 78 68 74 72 86 w=33.3765	(packages)
Mx 	14.8819
My 	134.9612
Push color	0.9333	0.9333	0.9333
Draw line	14.8819	108.6570	53.5748	52.6084
Pop color
Draw line	14.8819	108.6570	53.5748	0.4000
Draw line	14.8819	161.2654	53.5748	0.4000
Draw line	14.8819	108.6570	0.4000	53.0084
Draw line	68.4567	108.6570	0.4000	53.0084
Mx 	18.8819
My 	137.5320
T	55 88 72 86 71 68 92 w=30.6079	(Tuesday)
Mx 	68.4567
My 	117.4251
Draw line	68.4567	108.6570	42.8598	0.4000
Draw line	68.4567	126.1932	42.8598	0.4000
Draw line	68.4567	108.6570	0.4000	17.9361
Draw line	111.3165	108.6570	0.4000	17.9361
Mx 	75.0261
My 	119.9959
T	28 w=4.2231	(9)
Mx 	79.2492
T	29 w=2.0610	(:)
Mx 	81.3102
T	22 19 w=8.4463	(30)
Mx 	92.1349
T	36 48 w=12.6123	(AM)
Draw line	111.3165	108.6570	42.8598	0.4000
Draw line	111.3165	126.1932	42.8598	0.4000
Draw line	111.3165	108.6570	0.4000	17.9361
Draw line	154.1764	108.6570	0.4000	17.9361
Mx 	115.7743
T	20 20 w=8.4463	(11)
Mx 	124.2206
T	29 w=2.0610	(:)
Mx 	126.2817
T	19 19 w=8.4463	(00)
Mx 	137.1063
T	36 48 w=12.6123	(AM)
Mx 	68.4567
My 	134.9612
Draw line	68.4567	126.1932	42.8598	0.4000
Draw line	68.4567	143.7293	42.8598	0.4000
Draw line	68.4567	126.1932	0.4000	17.9361
Draw line	111.3165	126.1932	0.4000	17.9361
Mx 	72.9145
My 	137.5320
T	20 20 w=8.4463	(11)
Mx 	81.3608
T	29 w=2.0610	(:)
Mx 	83.4218
T	19 19 w=8.4463	(00)
Mx 	94.2464
T	36 48 w=12.6123	(AM)
Draw line	111.3165	126.1932	42.8598	0.4000
Draw line	111.3165	143.7293	42.8598	0.4000
Draw line	111.3165	126.1932	0.4000	17.9361
Draw line	154.1764	126.1932	0.4000	17.9361
Mx 	116.0995
T	20 21 w=8.4463	(12)
Mx 	124.5458
T	29 w=2.0610	(:)
Mx 	126.6069
T	19 19 w=8.4463	(00)
Mx 	137.4315
T	51 48 w=11.9619	(PM)
Mx 	68.4567
My 	152.4974
Draw line	68.4567	143.7293	42.8598	0.4000
Draw line	68.4567	161.2654	42.8598	0.4000
Draw line	68.4567	143.7293	0.4000	17.9361
Draw line	111.3165	143.7293	0.4000	17.9361
Mx 	73.2397
My 	155.0682
T	20 21 w=8.4463	(12)
Mx 	81.6860
T	29 w=2.0610	(:)
Mx 	83.7470
T	19 19 w=8.4463	(00)
Mx 	94.5716
T	51 48 w=11.9619	(PM)
Draw line	111.3165	143.7293	42.8598	0.4000
Draw line	111.3165	161.2654	42.8598	0.4000
Draw line	111.3165	143.7293	0.4000	17.9361
Draw line	154.1764	143.7293	0.4000	17.9361
Mx 	118.2111
T	24 w=4.2231	(5)
Mx 	122.4343
T	29 w=2.0610	(:)
Mx 	124.4953
T	19 19 w=8.4463	(00)
Mx 	135.3199
T	51 48 w=11.9619	(PM)
Mx 	154.1764
My 	121.8091
Draw line	154.1764	108.6570	128.5795	0.4000
Draw line	154.1764	134.9612	128.5795	0.4000
Draw line	154.1764	108.6570	0.4000	26.7042
Draw line	282.7559	108.6570	0.4000	26.7042
Mx 	158.1764
My 	124.3799
T	48 68 87 75 w=19.4502	(Math)
Mx 	180.0060
T	9 w=6.0029	(&)
Mx 	188.3883
T	68 71 89 68 81 70 72 71 w=35.1167	(advanced)
Mx 	225.8844
T	87 92 83 72 86 72 87 87 76 81 74 w=42.0864	(typesetting)
Mx 	154.1764
My 	148.1133
Draw line	154.1764	134.9612	128.5795	0.4000
Draw line	154.1764	161.2654	128.5795	0.4000
Draw line	154.1764	134.9612	0.4000	26.7042
Draw line	282.7559	134.9612	0.4000	26.7042
Mx 	158.1764
My 	150.6841
T	55 75 72 w=14.0801	(The)
Mx 	174.6390
T	76 81 87 72 85 81 68 79 86 w=33.2358	(internals)
Mx 	210.2575
T	82 73 w=7.3257	(of)
Mx 	219.9657
T	54 44 47 40 w=15.7148	(SILE)
Mx 	14.8819
My 	170.0335
Push color	0.9333	0.9333	0.9333
Draw line	14.8819	161.2654	53.5748	17.5361
Pop color
Draw line	14.8819	161.2654	53.5748	0.4000
Draw line	14.8819	178.8016	53.5748	0.4000
Draw line	14.8819	161.2654	0.4000	17.9361
Draw line	68.4567	161.2654	0.4000	17.9361
Mx 	18.8819
My 	172.6043
T	58 72 71 81 72 86 71 68 92 w=42.3369	(Wednesday)
Draw line	68.4567	161.2654	42.8598	0.4000
Draw line	68.4567	178.8016	42.8598	0.4000
Draw line	68.4567	161.2654	0.4000	17.9361
Draw line	111.3165	161.2654	0.4000	17.9361
Mx 	75.0261
T	27 w=4.2231	(8)
Mx 	79.2492
T	29 w=2.0610	(:)
Mx 	81.3102
T	19 19 w=8.4463	(00)
Mx 	92.1349
T	36 48 w=12.6123	(AM)
Draw line	111.3165	161.2654	42.8598	0.4000
Draw line	111.3165	178.8016	42.8598	0.4000
Draw line	111.3165	161.2654	0.4000	17.9361
Draw line	154.1764	161.2654	0.4000	17.9361
Mx 	116.0995
T	20 21 w=8.4463	(12)
Mx 	124.5458
T	29 w=2.0610	(:)
Mx 	126.6069
T	19 19 w=8.4463	(00)
Mx 	137.4315
T	51 48 w=11.9619	(PM)
Draw line	154.1764	161.2654	128.5795	0.4000
Draw line	154.1764	178.8016	128.5795	0.4000
Draw line	154.1764	161.2654	0.4000	17.9361
Draw line	282.7559	161.2654	0.4000	17.9361
Mx 	158.1764
T	54 44 47 40 w=15.7148	(SILE)
Mx 	176.2721
T	68 86 w=7.6069	(as)
Mx 	186.2599
T	68 81 w=9.0967	(an)
Mx 	197.7374
T	59 48 47 w=17.0771	(XML)
Mx 	217.1954
T	83 85 82 70 72 86 86 82 85 w=35.9912	(processor)
End page
Finish
